packages/maskbook/src/plugins/ITO/UI/RegionSelect.tsx

```diff
@@ -0,0 +1,232 @@
+import * as React from 'react'
+import { useState, useRef, useCallback, forwardRef, useImperativeHandle } from 'react'
+import type { InputBaseComponentProps } from '@material-ui/core'
+import { createStyles, makeStyles } from '@material-ui/core/styles'
+import { useDebounce } from 'react-use'
+import {
+    Typography,
+    Popover,
+    List,
+    ListItem,
+    ListItemText,
+    ListItemIcon,
+    Checkbox,
+    FormControl,
+    InputAdornment,
+    FilledInput,
+} from '@material-ui/core'
+
+import { Flags } from '../../../utils/flags'
+import { useI18N } from '../../../utils/i18n-next-ui'
+import { useRegionList } from '../hooks/useRegion'
+import type { RegionCode } from '../hooks/useRegion'
+import { usePortalShadowRoot } from '../../../utils/shadow-root/usePortalShadowRoot'
+
+export interface RegionSelectProps extends InputBaseComponentProps {
+    value: RegionCode[]
+    // onChange: React.ChangeEvent<{ value: RegionCode[] }> // TODO how to implement the ChangeEvent interface?
+    onRegionChange: (codes: RegionCode[]) => void
+}
+
+const useStyles = makeStyles((theme) =>
+    createStyles({
+        root: {
+            display: 'flex',
+            width: '100%',
+            height: '54px',
+            alignItems: 'center',
+            paddingLeft: theme.spacing(2),
+        },
+        allToggle: {
+            marginLeft: `-${theme.spacing(1)}`,
+        },
+        inputControl: {
+            display: 'flex',
+        },
+        inputRoot: {
+            backgroundColor: 'rgba(0, 0, 0, 0.04)',
+            '&:before': {
+                borderColor: 'rgba(0, 0, 0, 0.01)',
+            },
+        },
+        input: {
+            padding: `${theme.spacing(2)}`,
+        },
+        span: {
+            paddingLeft: theme.spacing(2),
+        },
+        options: {
+            maxHeight: 140,
+            overflow: 'auto',
+        },
+        display: {
+            padding: `${theme.spacing(0.5)} ${theme.spacing(2)}`,
+        },
+        item: {
+            padding: `0 ${theme.spacing(2)}`,
+        },
+    }),
+)
+
+// TODO fix TextField focus style
+export const RegionSelect = forwardRef(({ value = [], onRegionChange, ...props }: RegionSelectProps, ref) => {
+    const { t } = useI18N()
```

Comment:
hmm, can you explain why this component contains so much unusual stuff? (`useImperativeHandle`, "Ignore everything but left-click", ...)

Reply:
do some like the material ui [select](https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/Select/SelectInput.js#L209-L220), `useImperativeHandle` is for exporting fewer properties to parent. ` only left-click` is normal select behavior.